File: primos.py
import math

#primoBase = 22513583
#9007154355408959
#27093997468432753
#27244239137203931
#134286559156780091
#467737284162897641
#1629063613340779817
#2474644308342814631
#17786017647432497536200811937817548599
#5611651651165132132116513213651656184879849846541321321851987987984654131321165119847027
#158809173895446375831144734148401707861873653839436405804869463960548330057787962508639344452161267206832792283601459527829 
factor = 2 ** 120 

primoBase = 158809173895446375831144734148401707861873653839436405804869463960548330057787962508639344452161267206832792283601459527829

iteraciones = 1000
minimo = math.ceil(math.sqrt(primoBase))/2

print(f"este es el factor: {factor}")

primos = 0
noPrimos = 0
lastPrime = 0 

def isPrime(numero):
    for index in range (2, int(minimo/2)):
        if numero % index == 0:
            return False
    
    return numero




for index in range(1, iteraciones + 1):
    print(primoBase + ( factor * index ))
    primeCandidate = isPrime(primoBase + ( factor * index ) )

    if primeCandidate:
        lastPrime = primeCandidate
        print("este es primo", lastPrime)
        primos += 1
        break
    else:
        noPrimos += 1

    if index == iteraciones:
        print("con ", iteraciones, "el ultimo primo es:", lastPrime)

print("primos", primos)
print("no primos", noPrimos)

File: primos.ts
/// al parecer con numeros pequeños se conserva la cercania


//var factores = 17; // modificar aqui si quieres numeros mas grandes
var factor = 36; // modificar aqui si quieres numeros mas grandes
//var primoBase = 15024166879; // el numero primo base para comenzar la busqueda

/// numero maximo alcanzado en js
var primoBase = 9007199254740997; // el numero primo base para comenzar la busqueda
var iteraciones = 20; // encuentra cadenas de numeros primos con mas iteraciones

var primos = 0;
var noPrimos = 0;
var lastPrime = 0;

console.log("suma", 9007199254740996 + 3)

for (let i = 1; i <= iteraciones ; i++) {
    let primeCandidate = isPrime(9007199254740992 + 6);
    console.log("este es candidato", primoBase + (factor * i))

    if (primeCandidate) {
        lastPrime = primeCandidate;
        console.log("este es primo", lastPrime)

        primos += 1;
    } else {
        noPrimos += 1;
    }

    if (i == iteraciones) {
        console.log("con ", iteraciones, "el ultimo primo es:", lastPrime);
    }
}

console.log("primos", primos)
console.log("no primos", noPrimos)

console.log(factor.toString().length)
console.log(primoBase.toString().length)

function isPrime(numero: number) {
   //console.log("este es el numero", numero)
  
    for (var i = 2; i < 94906266/2; i++) {
        
        // implementar mejor manera de evaluar si es primo

        if (numero % i === 0) {
            return false;
        }

    }
    return numero;;

}
